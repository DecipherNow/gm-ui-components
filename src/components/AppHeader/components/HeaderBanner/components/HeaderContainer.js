import styled from "styled-components";

import { contrastColor, spacingScale } from "style/styleFunctions";
import { COLOR_BLACK } from "style/styleVariables";

const HeaderContainer = styled.div`
  color: ${contrastColor(COLOR_BLACK, 1)};
  padding: ${spacingScale(1)} ${spacingScale(2)} ${spacingScale(1)};
  transition: all 0.1s ease;

  @media all and (min-width: 1000px) {
    padding: ${spacingScale(2)} ${spacingScale(2)} ${spacingScale(1)};
  }
`;

export default HeaderContainer;
