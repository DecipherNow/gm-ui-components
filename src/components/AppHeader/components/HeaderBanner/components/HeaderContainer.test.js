import React from "react";
import { shallow } from "enzyme";
import HeaderContainer from "./HeaderContainer";

const theme = {
  COLOR_BACKGROUND_C: "blue"
};

describe("HeaderContainer", () => {
  it("matches snapshot", () => {
    const aHeaderContainer = shallow(<HeaderContainer theme={theme} />);
    expect(aHeaderContainer).toMatchSnapshot();
  });
});
