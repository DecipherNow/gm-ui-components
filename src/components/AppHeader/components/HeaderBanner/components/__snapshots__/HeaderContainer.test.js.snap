// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "HeaderContainer-sc-1e4y2wi-0",
        "isStatic": false,
        "rules": Array [
          "color:#fff;padding:",
          [Function],
          "px",
          " ",
          [Function],
          "px",
          " ",
          [Function],
          "px",
          ";transition:all 0.1s ease;@media all and (min-width:1000px){padding:",
          [Function],
          "px",
          " ",
          [Function],
          "px",
          " ",
          [Function],
          "px",
          ";}",
        ],
      },
      "displayName": "HeaderContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "HeaderContainer-sc-1e4y2wi-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
